testing/src/main/java/org/oppia/android/testing/TextInputAction.kt

```diff
@@ -0,0 +1,50 @@
+package org.oppia.android.testing
+
+import android.content.Context
+import android.view.View
+import androidx.annotation.StringRes
+import com.google.android.material.textfield.TextInputLayout
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.TypeSafeMatcher
+import javax.inject.Inject
+
+/**
+ * Action for matching error text and performing other actions on TextInputLayout in a test infrastructure-specific way.
+ */
+class TextInputAction @Inject constructor(val context: Context) {
+  // TODO(#1720): Move this to a companion object & use a test-only singleton injector to retrieve.
+
+  /**
+   * @param expectedErrorTextId is the string id which needs to match.
+   *
+   * @returns a [Matcher] that matches the specified string [expectedErrorTextId] with the view's error text.
+   */
+  fun hasErrorText(@StringRes expectedErrorTextId: Int): Matcher<View> {
+    return object : TypeSafeMatcher<View>() {
```

Comment:
Shouldn't the type being passed in be ``TextInputLayout`` so that we're properly matching against that type?

Ditto for the other method below.

Reply:
@BenHenning 
Its How the Function is used.It will cause to change many things to use as the function parameter

![image](https://user-images.githubusercontent.com/60266468/110094208-9496c100-7dc1-11eb-8910-6493eefc4f7d.png)

Reply:
@BenHenning PTAL

Reply:
@BenHenning 
**this is how we use the function to match.**
**Should I change to pass as parameter.**
![image](https://user-images.githubusercontent.com/60266468/110760054-4d527980-8274-11eb-81c6-15324900883e.png)

Reply:
Sorry, what I meant was: could we instead use ``TypeSafeMatcher<TextInputLayout>`` instead of ``TypeSafeMatcher<View>``? I think the former will allow us to simplify the code here since we wouldn't then need to cast the view to a ``TextInputLayout`` (``TypeSafeMatcher`` does that for us).

Reply:
@BenHenning 
Do you mean to use like this
![image](https://user-images.githubusercontent.com/60266468/111362205-8778ac80-86b4-11eb-9bfd-7606d02be5ae.png)